recipes/libxml2/all/conanfile.py

```diff
@@ -230,34 +232,55 @@ def package(self):
             # remove redundant libraries to avoid confusion
             if not self.options.shared:
                 os.unlink(os.path.join(self.package_folder, "bin", "libxml2.dll"))
-            os.unlink(os.path.join(self.package_folder, 'lib', 'libxml2_a_dll.lib'))
-            os.unlink(os.path.join(self.package_folder, 'lib', 'libxml2_a.lib' if self.options.shared else 'libxml2.lib'))
+            os.unlink(os.path.join(self.package_folder, "lib", "libxml2_a_dll.lib"))
+            os.unlink(os.path.join(self.package_folder, "lib", "libxml2_a.lib" if self.options.shared else "libxml2.lib"))
 
-            pdb_files = glob.glob(os.path.join(self.package_folder, 'bin', '*.pdb'), recursive=True)
+            pdb_files = glob.glob(os.path.join(self.package_folder, "bin", "*.pdb"), recursive=True)
             for pdb in pdb_files:
                 os.unlink(pdb)
 
-        tools.rmdir(os.path.join(self.package_folder, 'share'))
-        tools.rmdir(os.path.join(self.package_folder, 'lib', 'cmake'))
-        tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
 
     def package_info(self):
         if self._is_msvc:
-            self.cpp_info.libs = ['libxml2' if self.options.shared else 'libxml2_a']
+            self.cpp_info.components["xml2lib"].libs = ["libxml2" if self.options.shared else "libxml2_a"]
         else:
-            self.cpp_info.libs = ['xml2']
-        self.cpp_info.includedirs.append(os.path.join("include", "libxml2"))
+            self.cpp_info.components["xml2lib"].libs = ["xml2"]
+        self.cpp_info.components["xml2lib"].includedirs.append(os.path.join("include", "libxml2"))
         if not self.options.shared:
-            self.cpp_info.defines = ["LIBXML_STATIC"]
-        if self.options.include_utils:
-            bindir = os.path.join(self.package_folder, "bin")
-            self.output.info("Appending PATH environment variable: {}".format(bindir))
-            self.env_info.PATH.append(bindir)
+            self.cpp_info.components["xml2lib"].defines = ["LIBXML_STATIC"]
 
-        if self.settings.os == "Linux" or self.settings.os == "Macos":
-            self.cpp_info.system_libs.append('m')
+        if self.settings.os == "Linux":
+            self.cpp_info.components["xml2lib"].system_libs.append("m")
         if self.settings.os == "Windows":
-            self.cpp_info.system_libs.append('ws2_32')
+            self.cpp_info.components["xml2lib"].system_libs.append("ws2_32")
+        if self.options.threads:
+            if self.settings.os == "Linux":
+                self.cpp_info.components["xml2lib"].system_libs.append("pthread")
```

Comment:
```suggestion
        if self.options.threads and if self.settings.os == "Linux"
                self.cpp_info.components["xml2lib"].system_libs.append("pthread")
```

Reply:
I can't review the all block, but I would simplify `system_libs` part with:

```python
        if self.settings.os == "Linux":
            self.cpp_info.components["xml2lib"].system_libs = ["m"]
            if self.options.threads:
                self.cpp_info.components["xml2lib"].system_libs.append("pthread")
        elif self.settings.os == "Windows":
            self.cpp_info.components["xml2lib"].system_libs = ["ws2_32"]
```

Reply:
also ws2_32 is only required if ftp and http options are True.
In Makefile.mingw:

```
ifeq ($(WITH_FTP),1)
LIBS += -lwsock32 -lws2_32
endif 
ifeq ($(WITH_HTTP),1)
LIBS += -lwsock32 -lws2_32
endif 
```

Reply:
Thanks guys, @SpaceIm 's pr fixed this.